led_strip/src/led_strip_rmt_dev.c

```diff
@@ -37,6 +37,7 @@ static esp_err_t led_strip_rmt_set_pixel(led_strip_t *strip, uint32_t index, uin
 {
     led_strip_rmt_obj *rmt_strip = __containerof(strip, led_strip_rmt_obj, base);
     ESP_RETURN_ON_FALSE(index < rmt_strip->strip_len, ESP_ERR_INVALID_ARG, TAG, "index out of maximum number of LEDs");
+    ESP_RETURN_ON_FALSE(rmt_strip->bytes_per_pixel == 3, ESP_ERR_INVALID_ARG, TAG, "wrong LED pixel format, expected 3 bytes per pixel");
```

Comment:
the previous logic doesn't forbid the user from using this function on an RGBW format led strip (check the code below, if the white element is not specified. we set it to zero).

Reply:
This logic should also be applied to the spi interfaced led strip.